lib/engine/phase.rb

```diff
@@ -107,12 +108,13 @@ def rust_trains!(train, entity)
         next unless should_rust
 
         rusted_trains << t.name
+        owners << t.owner.name
         entity.rusted_self = true if entity && entity == t.owner
         t.rust!
       end
 
       @log << "-- Event: #{obsolete_trains.uniq.join(', ')} trains are obsolete --" if obsolete_trains.any?
-      @log << "-- Event: #{rusted_trains.uniq.join(', ')} trains rust --" if rusted_trains.any?
+      @log << "-- Event: #{rusted_trains.uniq.join(', ')} trains rust (#{owners.group_by(&:itself).map{|c,t| "#{c} x#{t.count}"}.join(', ')}) --" if rusted_trains.any?       
```

Comment:
owners = Hash.new(0)

owners[t.owner.name] += 1

owners.map { |name, count| "#{...}"...

Reply:
also your lines are too long and the formatting is off, you can find them locally with 

docker-compose exec rack rake